File: app/src/main/java/com/gomart/guildbuddy/module/GuildModule.java
package com.gomart.guildbuddy.module;

import android.content.Context;

import com.gomart.guildbuddy.model.Guild;
import com.gomart.guildbuddy.ui.presenter.GuildPresenter;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by glaubermartins on 2016-12-05.
 */

@Module
public class GuildModule {

    private Guild guild;
    private Context context;

    public GuildModule(Context context, Guild guild){
        this.guild = guild;
        this.context = context;
    }

    @Singleton
    @Provides
    GuildPresenter provideGuildPresenter(){
        return new GuildPresenter(context, guild);
    }
}

File: app/src/main/java/com/gomart/guildbuddy/di/GuildModule.kt
package com.gomart.guildbuddy.di

import com.gomart.guildbuddy.data.AppDatabase
import com.gomart.guildbuddy.data.GuildDao
import com.gomart.guildbuddy.network.services.GuildService
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import retrofit2.Retrofit
import javax.inject.Singleton

/**
 *   Created by gmartins on 2020-08-28
 *   Description:
 */
@Module
@InstallIn(SingletonComponent::class)
object GuildModule {
    @Singleton
    @Provides
    fun provideGuildService(@RetrofitClient retrofit: Retrofit): GuildService =
            retrofit.create(GuildService::class.java)

    @Singleton
    @Provides
    fun provideGuildDao(@DatabaseClient appDatabase: AppDatabase): GuildDao =
            appDatabase.guildDao()
}